### PR TITLE
Modernize `dactyl_manuform/dmote` keyboard

### PR DESCRIPTION
**Pull request by dunk2k**
## Description
- `dmote` dir moved up one level
    - alias added
- folder structure changed to match Keyboard PR guidance
    - file names changed to match folder structure
    - layout macro name renamed
- `layout and keymaps reordered/modernised to be uniform with other split keyboards 
- `readme.md` updated

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* n/a

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/dmote/readme.md [hunk 1]
```diff
@@ -1,16 +1,29 @@
-DMOTE
-======
+# DMOTE
+
+![DMOTE original](https://camo.githubusercontent.com/09e9ce3409bf5d0385e8616639d6295851546cca0743501d8e66615467c0eed2/687474703a2f2f76696b746f722e65696b6d616e2e73652f696d6167652f646d6f74652d322d746f702d646f776e2d766965772f646973706c6179)
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/dmote/readme.md:3]: Should be hosted on imgur.

Comment (dunk2k) [reply keyboards/handwired/dactyl_manuform/dmote/readme.md]: Image link removed

**Comment by drashna** [comment]
```sh
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout qmk-develop
```

**Review by drashna — APPROVED** [review]

File: data/mappings/keyboard_aliases.hjson [hunk 2]
```diff
@@ -199,6 +199,9 @@
     "handwired/dactyl_manuform/6x6_kinesis": {
         "target": "handwired/dactyl_kinesis"
     },
+    "handwired/dactyl_manuform/dmote/62key": {
+        "target": "handwired/dmote"
+    },
```
Comment (fauxpark) [inline data/mappings/keyboard_aliases.hjson:204]:
```suggestion
    },
```

**Review by fauxpark — APPROVED** [review]

**Pull request merged**